### PR TITLE
Refactor migration route handling and add tests

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

Simplify how migration endpoints are served and add tests.

### Testing

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of WordPress

### Checklist

* [x] All existing and new tests complete without errors
* [x] All code quality tools/guidelines in the [Contribution guide](CONTRIBUTION.md) have been run/followed
* [x] All active GitHub CI checks have passed

### CODE REVIEW

File: lib/WP_Auth0_Routes.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function init() {
 		add_action( 'parse_request', array( $this, 'custom_requests' ) );
 	}
 
-	public function setup_rewrites( $force_ws = false ) {
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: Allowed in PHP. Parameter was not used and anyone calling this with a parameter will not get an error.

File: lib/WP_Auth0_Routes.php [hunk 2]
```diff
@@ -157,38 +145,13 @@ protected function getAuthorizationHeader() {
...
-			return $this->error_return_array( 403 );
-		}
-
-		// IP filtering is on and incoming IP address does not match filter.
-		if ( $this->a0_options->get( 'migration_ips_filter' ) ) {
-			$allowed_ips = $this->a0_options->get( 'migration_ips' );
-			if ( ! $this->ip_check->connection_is_valid( $allowed_ips ) ) {
-				return $this->error_return_array( 401 );
-			}
+		$code = $this->check_endpoint_access_error();
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: This call handles all the checks above.

File: lib/WP_Auth0_Routes.php [hunk 3]
```diff
@@ -217,35 +180,13 @@ protected function migration_ws_login() {
...
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: This call handles all the checks above.

File: lib/WP_Auth0_Routes.php [hunk 4]
```diff
@@ -217,35 +180,13 @@ protected function migration_ws_login() {
...
-			}
-
-			if ( ! $this->valid_token( $authorization ) ) {
-				throw new Exception( __( 'Invalid token', 'wp-auth0' ), 401 );
-			}
-
-			if ( ! isset( $_POST['username'] ) ) {
-				throw new Exception( __( 'Username is required', 'wp-auth0' ) );
-			}
+			$this->check_endpoint_request();
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: This call handles all the checks above.

File: lib/WP_Auth0_Routes.php [hunk 5]
```diff
@@ -157,38 +145,13 @@ protected function getAuthorizationHeader() {
...
-			}
-
-			if ( empty( $_POST['username'] ) ) {
-				throw new Exception( __( 'Username is required', 'wp-auth0' ) );
-			}
-
-			if ( empty( $_POST['password'] ) ) {
-				throw new Exception( __( 'Password is required', 'wp-auth0' ) );
-			}
+			$this->check_endpoint_request( true );
```
Comment (joshcanhelp) [inline lib/WP_Auth0_Routes.php]: This call handles all the checks above.

File: lib/WP_Auth0_Routes.php [hunk 6]
```diff
@@ -272,17 +213,62 @@ protected function migration_ws_get_user() {
...
+	/**
+	 * Check the incoming request for token and required data.
+	 *
+	 * @param bool $require_password - True to check for a POSTed password, false to ignore.
+	 *
+	 * @throws Exception
+	 */
+	private function check_endpoint_request( $require_password = false ) {
+		$authorization = $this->getAuthorizationHeader();
+		$authorization = trim( str_replace( 'Bearer ', '', $authorization ) );
```
Comment (cocojoe) [inline lib/WP_Auth0_Routes.php]: Feels like the sort of thing that might be nice in a tested helper method.

**Pull request merged**